File: Assignment 2_code files/account.cpp
#include <iostream>
#include <functional>
using namespace std;

function<int(int)> makeAccount(int balance) {
    return [balance](int amount) mutable {
        balance += amount;
        return balance;
    };
}

int main() {
    auto acct = makeAccount(100);
    cout << acct(50) << endl;   // 150
    // int v = "text";  // won't compile: static typing
}

File: Assignment 2_code files/account.py
def make_account(balance):
    def deposit(amount):
        nonlocal balance
        balance += amount
        return balance
    return deposit

acct = make_account(100)
print(acct(50))            # 150
value = 10
value = "text"             # dynamic typing: rebinds freely
print(value)